frontend/public/components/graphs/area.tsx

```diff
@@ -35,13 +39,35 @@ export const AreaChart: React.FC<AreaChartProps> = ({
   query,
   theme = getCustomTheme(ChartThemeColor.blue, ChartThemeVariant.light, areaTheme),
   tickCount = DEFAULT_TICK_COUNT,
+  timespan,
   title,
   xAxis = true,
   yAxis = true,
 }) => {
   const [containerRef, width] = useRefWidth();
+  const [timestamp, setTimestamp] = React.useState(null);
+  React.useEffect(() => {
+    setTimestamp(Date.now());
+  }, [data])
+
   const getLabel = ({x, y}) => `${humanize(y).string} at ${formatDate(x)}`;
-  const container = <ChartVoronoiContainer voronoiDimension="x" labels={getLabel} />;
+  const container = <ChartVoronoiContainer labels={getLabel} voronoiDimension="x" />;
+  const max = data.length ? _.maxBy(data, 'y') : {};
+
+  // Force x-domain to show entire timespan
+  // Force y-domain to be a reasonable range for a single data point or if all data values are 0
+  const domain: DomainPropType = timestamp && {
+    ...timespan && { x: [
+      timestamp - timespan,
```

Comment:
Is it safe to assume that the chart is showing data from the current time, or could this component show some other time range?

Reply:
It is possible that the graph could cover some other timespan. The data is independent of timespan, so it's possible that you could pass in data from the past, then pass in a timespan prop that would cause that data to be out of range. Maybe we need `startTime` and `endTime`, or `endTime` and `timespan`.